packages/api/lambdas/sf-semaphore-down.js

```diff
@@ -62,34 +71,39 @@ async function decrementQueueSemaphore(queueName) {
 }
 
 /**
- * Filter workflow messages from SNS to prepare array of semaphore decrement tasks.
+ * Handle Cloudwatch event and decrement semaphore, if necessary.
  *
- * @param {Object} event - incoming message from SNS
- * @returns {Array<Promise>} - Array of promises for semaphore decrement operations
+ * @param {Object} event - incoming event from Cloudwatch
+ * @returns {Promise}
  */
-function getSemaphoreDecrementTasks(event) {
-  return get(event, 'Records', [])
-    // Skip if this record is not from SNS or if the SNS message is empty
-    .filter((record) => has(record, 'Sns.Message') && !isNil(record.Sns.Message))
-    .map((record) => JSON.parse(record.Sns.Message))
-    .filter((message) => isDecrementMessage(message))
-    .map((message) => getQueueName(message))
-    .map((queueName) => decrementQueueSemaphore(queueName));
+async function handleSemaphoreDecrementTask(event) {
+  if (!isSfExecutionEvent(event)) {
+    return Promise.resolve();
+  }
+
+  const status = get(event, 'detail.status');
+  const output = get(event, 'detail.output', '{}');
+  const message = JSON.parse(output);
```

Comment:
It seems possible that `output` could just be a regular string or some other value that is not a JSON string. I think this needs to be more defensive